src/main/java/appeng/helpers/PatternHelper.java

```diff
@@ -85,7 +85,13 @@ public PatternHelper( final ItemStack is, final World w )
 
 		for( int x = 0; x < inTag.tagCount(); x++ )
 		{
-			final ItemStack gs = new ItemStack( inTag.getCompoundTagAt( x ) );
+			NBTTagCompound ingredient = inTag.getCompoundTagAt( x );
+			final ItemStack gs = new ItemStack( ingredient );
+
+			if( !ingredient.hasNoTags() && gs.isEmpty() )
+			{
+				throw new IllegalArgumentException( "No pattern here!" );
```

Comment:
the inTag list is not compressed, so its valid to create an empty itemstack gs. But that is indistinguishable from trying to create an ItemStack for an non-existent item from the NBT. As far as I have seen it, serializing an empty ItemStack results in an NBTCompound without tags. Thus, the if validates, that gs is empty and that the input NBT doesn't denote an empty stack.

So in fact, any ingredient ending there can't be parsed to a valid pattern.